cloud/services/virtualmachines/virtualmachines.go

```diff
@@ -126,6 +126,12 @@ func (s *Service) Reconcile(ctx context.Context, spec interface{}) error {
 			HardwareProfile: &compute.HardwareProfile{
 				VMSize: compute.VirtualMachineSizeTypes(vmSpec.Size),
 			},
+			DiagnosticsProfile: &compute.DiagnosticsProfile{
```

Comment:
These are all great changes but it's too much for one PR, can you please split 1) vm diagnostics 2) NetworkDescriber/ClusterDescriber changes 3) AzureManagedControlPlane etc.  in multiple PRs?

Reply:
yeah, i'm mostly trying to see if I can get it working first // share some of the changes to patterns that have been helpful trying to make this work for different scenarios. I definitely won't try to merge this as one PR, and I found a few small bits here and there to merge as one-off fixes besides the main functional stuff.

Reply:
also vm diagnostics is hardcoded, that was me debugging. once I figure out which order to tackle all these properly I can assign the relevant issues.